src/core/raster/qgsrasterlayerrenderer.cpp

```diff
@@ -227,6 +227,16 @@ QgsRasterLayerRenderer::QgsRasterLayerRenderer( QgsRasterLayer *layer, QgsRender
   QgsRasterRenderer *rasterRenderer = mPipe->renderer();
   if ( rasterRenderer && !( rendererContext.flags() & QgsRenderContext::RenderPreviewJob ) )
     layer->refreshRendererIfNeeded( rasterRenderer, rendererContext.extent() );
+
+  if ( layer && mPipe->provider() && mPipe->provider()->temporalCapabilities() )
```

Comment:
```suggestion
  if ( renderContext.isTemporal() && mPipe->provider()->temporalCapabilities() )
```
(we are confident at this stage that layer && provider exist!)